migration/handler_test.go

```diff
@@ -0,0 +1,120 @@
+package migration
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/store"
+	"github.com/iov-one/weave/weavetest"
+	"github.com/iov-one/weave/weavetest/assert"
+)
+
+func TestSchemaMigratingHandler(t *testing.T) {
+	const thisPkgName = "testpkg"
+
+	reg := newRegister()
+
+	reg.MustRegister(1, &MyMsg{}, NoModification)
+	reg.MustRegister(2, &MyMsg{}, func(db weave.ReadOnlyKVStore, m Migratable) error {
+		msg := m.(*MyMsg)
+		msg.Content += " m2"
+		return msg.err
+	})
+	reg.MustRegister(3, &MyMsg{}, func(db weave.ReadOnlyKVStore, m Migratable) error {
+		panic("not implemented")
+	})
+
+	db := store.MemStore()
+
+	schema := NewSchemaBucket()
+	if _, err := schema.Create(db, &Schema{Metadata: &weave.Metadata{Schema: 1}, Pkg: thisPkgName, Version: 1}); err != nil {
+		t.Fatalf("cannot register schema version: %s", err)
+	}
+
+	handler := SchemaMigratingHandler(thisPkgName, &weavetest.Handler{})
+	// Use custom register reference so that our test is not polluted by
+	// extenrnal registrations.
+	handler.(*schemaMigratingHandler).migrations = reg
```

Comment:
I would suggest using some cleaner mock setup like I proposed for Bucket.
If `SchemaMigratingHandler` returns `*schemaMigratingHandler` not the interface, then you have access to those extra methods, at least in test code.

Reply:
I did not want to expose this handler implementations, that's why I'm returning `weave.Handler`. Making this a public method would require `register` to be public as well. 

Instead I have created a private function to set this.  https://github.com/iov-one/weave/pull/550/commits/88270383a81e3290ce2993b3fea9beaa8579bfc5#diff-1c3925c3634d36476b64497288c1c6b3R122